### PR TITLE
envoy: Rebase to get gRPC proxy responses.

### PR DESCRIPTION
**Pull request by jrajahalme**
Rebase to the upstream Envoy commit that merged the gRPC responses to gRPC requests.

Signed-off-by: Jarno Rajahalme <jarno@covalent.io>

### CODE REVIEW

File: envoy/WORKSPACE [hunk 1]
```diff
@@ -38,8 +37,8 @@ load("@com_lyft_protoc_gen_validate//bazel:go_proto_library.bzl", "go_proto_repo
 go_proto_repositories(shared=0)
 go_rules_dependencies()
 go_register_toolchains()
-load("@io_bazel_rules_go//proto:def.bzl", "proto_register_toolchains")
-proto_register_toolchains()
+# load("@io_bazel_rules_go//proto:def.bzl", "proto_register_toolchains")
```
Comment (rlenglet) [inline envoy/WORKSPACE]: I don't understand this particular change.
If that's legit, please delete the lines instead of commenting them.

**Review by rlenglet — APPROVED** [review]

**Comment by rlenglet** [comment]
test-me-please

**Comment by rlenglet** [comment]
I assume you're going to generate the Vagrant box and Docker image, and then update the versions in this PR?

**Comment by rlenglet** [comment]
@jrajahalme Please check that `cd envoy ; make istio-envoy` still works. You may need to bump the version of `istio/proxy` to make the Istio filters compile.

File: envoy/WORKSPACE (hunk 1 above)
Comment (jrajahalme) [reply envoy/WORKSPACE]: Right, I did not see these lines in upstream WORKSPACE files, so decided to comment them out to see if they are (still) required. Will remove, as envoy builds in this PR.

**Comment by jrajahalme** [comment]
Still need to figure out how to use go dep as golang/protobuf needs an update to a newer version. I guess this is due to us using latest protoc-gen-go, but a fixed version of golang/protobuf :-(

**Comment by jrajahalme** [comment]
Istio filters need some updates too, haven't checked yet if upstream updates exist.

**Comment by rlenglet** [comment]
@jrajahalme I'd use Istio Proxy's master HEAD (https://github.com/istio/proxy/commit/709a36650b09f6a46e768bcc0c155609c97f8876). Its Istio ENVOY SHA is `2b2c299144600fb9e525d21aabf39bf48e64fb1f`.
https://github.com/istio/proxy/blob/master/WORKSPACE#L41

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Comment by ianvernon** [comment]
>  I'd use Istio Proxy's master HEAD (istio/proxy@709a366). Its Istio ENVOY SHA is 2b2c299144600fb9e525d21aabf39bf48e64fb1f.
https://github.com/istio/proxy/blob/master/WORKSPACE#L41

Is it not possible for us to use a versioned release? I don't like using commits off of master for any dependency.

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
@rlenglet Right, Istio uses Envoy SHA `2b2c299144600fb9e525d21aabf39bf48e64fb1f`, but that is too old, it does not have the gRPC fixes yet. Need to figure out if changes since `2b2c299144600fb9e525d21aabf39bf48e64fb1f` break Istio filters or not.

**Comment by rlenglet** [comment]
@jrajahalme I'm checking right now. Will let you know.

**Comment by rlenglet** [comment]
It breaks the Istio filters. :( I'll try to build Envoy for the Istio proxy images with the ENVOY SHA from Istio.

**Comment by jrajahalme** [comment]
@rlenglet Most likely the needed changes are similar namespace changes that are done in this PR to our filters. Should be simple to fix, would you like to do that & propose upstream PR to Istio?

**Comment by jrajahalme** [comment]
test-me-please

**Comment by jrajahalme** [comment]
@rlenglet Issued https://github.com/istio/proxy/pull/1758

**Comment by jrajahalme** [comment]
CI passed: https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/3385 :-)

**Comment by jrajahalme** [comment]
@rlenglet Wonder if I need to rebase this to master before this can be merged, or can it be "rebase and merged" without running CI again?

**Pull request merged**